linkcheck/director/aggregator.py

```diff
@@ -79,11 +79,12 @@ def visit_loginurl(self):
         if not user and not password:
             raise LinkCheckerError(
                 "loginurl is configured but neither user nor password are set")
-        session = requests.Session()
-        # XXX user-agent header
-        # XXX timeout
+        session = new_request_session(self.config, self.cookies)
         log.debug(LOG_CHECK, "Getting login form %s", url)
-        response = session.get(url)
+        kwargs = dict(timeout=self.config["timeout"])
+        # XXX: proxy?  sslverify?  can we reuse HttpUrl.get_request_kwargs()
+        # somehow?
+        response = session.get(url, **kwargs)
```

Comment:
I am tempted to switch from requests to [httpx](https://www.python-httpx.org/), once httpx 1.0 becomes available.

I looked at the API docs and it seems requests-like, but nicer.

Reply:
curious to discuss this more, but i'd say we should cross that bridge when we get there... what would be the biggest advantage for switching in linkchecker?